diff --git a/bazel/config/configs.bzl b/bazel/config/configs.bzl
index b294c40d49e..19f501ff7c9 100644
--- a/bazel/config/configs.bzl
+++ b/bazel/config/configs.bzl
@@ -270,6 +270,20 @@ build_otel = rule(
     build_setting = config.bool(flag = True),
 )
 
+# =========
+# grpc
+# =========
+
+build_grpc_provider = provider(
+    doc = """Enable building grpc and protobuf compiler. This has no effect on non-linux operating systems.""",
+    fields = ["enabled"],
+)
+
+build_grpc = rule(
+    implementation = lambda ctx: build_grpc_provider(enabled = ctx.build_setting_value),
+    build_setting = config.bool(flag = True),
+)
+
 # =========
 # sanitize
 # =========
