diff --git a/bazel/mongo_src_rules.bzl b/bazel/mongo_src_rules.bzl
index f7d014d7dc5..c2c4a640b5a 100644
--- a/bazel/mongo_src_rules.bzl
+++ b/bazel/mongo_src_rules.bzl
@@ -232,6 +232,7 @@ RE_ENABLE_DISABLED_3RD_PARTY_WARNINGS_FEATURES = select({
 })
 
 MONGO_GLOBAL_SRC_DEPS = [
+    "//src:core_headers_library",
     "//src/third_party/abseil-cpp:absl_base",
     "//src/third_party/boost:boost_system",
     "//src/third_party/croaring:croaring",
