diff --git a/bazel/config/BUILD.bazel b/bazel/config/BUILD.bazel
index 556e8628aef..1a41efe8c50 100644
--- a/bazel/config/BUILD.bazel
+++ b/bazel/config/BUILD.bazel
@@ -341,7 +341,7 @@ selects.config_setting_group(
 
 mongo_toolchain_version(
     name = "mongo_toolchain_version",
-    build_setting_default = "v5",
+    build_setting_default = "v4",
 )
 
 config_setting(
@@ -351,13 +351,6 @@ config_setting(
     },
 )
 
-config_setting(
-    name = "mongo_toolchain_v5",
-    flag_values = {
-        "//bazel/config:mongo_toolchain_version": "v5",
-    },
-)
-
 # ==========
 # opt
 # ==========
diff --git a/bazel/config/configs.bzl b/bazel/config/configs.bzl
index 358f6d3b50e..1a0b73a36ca 100644
--- a/bazel/config/configs.bzl
+++ b/bazel/config/configs.bzl
@@ -26,10 +26,10 @@ compiler_type = rule(
 # mongo_toolchain_version
 # =========
 
-mongo_toolchain_version_values = ["v5"]
+mongo_toolchain_version_values = ["v4"]
 
 mongo_toolchain_version_provider = provider(
-    doc = "Select the mongo toolchain version (e.g.: v5)",
+    doc = "Select the mongo toolchain version (e.g.: v4)",
     fields = {"mongo_toolchain_version": "Choose one of " + ".".join(mongo_toolchain_version_values)},
 )
 
diff --git a/bazel/mongo_script_rules.bzl b/bazel/mongo_script_rules.bzl
index f4ed32b0fb7..cd908fd1b25 100644
--- a/bazel/mongo_script_rules.bzl
+++ b/bazel/mongo_script_rules.bzl
@@ -3,8 +3,8 @@
 
 load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
 
-MONGO_TOOLCHAIN_V4_PATH = "/opt/mongodbtoolchain/v4"
-MONGO_TOOLCHAIN_V5_PATH = "external/mongo_toolchain_v5/v5"
+MONGO_TOOLCHAIN_V4_PATH = "external/mongo_toolchain_v4/v4"
+
 
 def _py_cxx_wrapper(*, python_path, toolchain_path, python_interpreter, main_py):
     return "\n".join([
@@ -81,7 +81,7 @@ py_cxx_test = rule(
 def mongo_toolchain_py_cxx_test(**kwargs):
     py_cxx_test(
         toolchain_path = select({
-            "//bazel/config:mongo_toolchain_v5": MONGO_TOOLCHAIN_V5_PATH,
+            "//bazel/config:mongo_toolchain_v4": MONGO_TOOLCHAIN_V4_PATH,
             "//conditions:default": MONGO_TOOLCHAIN_V4_PATH,
         }),
         target_compatible_with = ["@//bazel/platforms:use_mongo_toolchain"],
diff --git a/bazel/toolchains/cc/mongo_linux/mongo_linux_cc_toolchain_config.bzl b/bazel/toolchains/cc/mongo_linux/mongo_linux_cc_toolchain_config.bzl
index 54e3ce5f675..1e492ddbbd4 100644
--- a/bazel/toolchains/cc/mongo_linux/mongo_linux_cc_toolchain_config.bzl
+++ b/bazel/toolchains/cc/mongo_linux/mongo_linux_cc_toolchain_config.bzl
@@ -789,8 +789,6 @@ def _impl(ctx):
                 actions = all_compile_actions,
                 flag_groups = [flag_group(flags = [
                     "-Wno-deprecated-declarations",
-                    "-Wno-deprecated-non-prototype",
-                    "-Wno-missing-template-arg-list-after-template-kw",
                     "-Wno-sign-compare",
                 ])],
             ),
diff --git a/bazel/toolchains/cc/mongo_linux/mongo_toolchain.bzl b/bazel/toolchains/cc/mongo_linux/mongo_toolchain.bzl
index 4fce3d046ef..482fcd44ffb 100644
--- a/bazel/toolchains/cc/mongo_linux/mongo_toolchain.bzl
+++ b/bazel/toolchains/cc/mongo_linux/mongo_toolchain.bzl
@@ -50,7 +50,7 @@ toolchain_download = repository_rule(
             doc = "Host architecture.",
         ),
         "version": attr.string(
-            values = ["v5"],
+            values = ["v4"],
             doc = "Mongodbtoolchain version.",
             mandatory = True,
         ),
@@ -67,13 +67,13 @@ toolchain_download = repository_rule(
 
 def setup_mongo_toolchains():
     toolchain_download(
-        name = "mongo_toolchain_v5",
-        version = "v5",
-        flags_tpl = "//bazel/toolchains/cc/mongo_linux:mongo_toolchain_flags_v5.bzl",
+        name = "mongo_toolchain_v4",
+        version = "v4",
+        flags_tpl = "//bazel/toolchains/cc/mongo_linux:mongo_toolchain_flags_v4.bzl",
     )
 
     native.register_toolchains(
-        "@mongo_toolchain_v5//:all",
+        "@mongo_toolchain_v4//:all",
     )
 
 # Defines aliases for key targets inside the toolchain the user has chosen via
@@ -95,7 +95,7 @@ def setup_mongo_toolchain_aliases():
         selects[target] = {}
 
     for target in toolchain_targets:
-        selects[target]["//bazel/config:mongo_toolchain_v5"] = "@mongo_toolchain_v5//:{}".format(target)
+        selects[target]["//bazel/config:mongo_toolchain_v4"] = "@mongo_toolchain_v4//:{}".format(target)
 
     for target, local_alias in toolchain_targets.items():
         native.alias(
diff --git a/bazel/toolchains/cc/mongo_linux/mongo_toolchain_flags_v4.bzl b/bazel/toolchains/cc/mongo_linux/mongo_toolchain_flags_v4.bzl
new file mode 100644
index 00000000000..38ea0805976
--- /dev/null
+++ b/bazel/toolchains/cc/mongo_linux/mongo_toolchain_flags_v4.bzl
@@ -0,0 +1,33 @@
+COMMON_LINK_FLAGS = [
+    "external/mongo_toolchain_v4/stow/gcc-v4/lib/gcc/{arch}-mongodb-linux/11.3.0",
+    "external/mongo_toolchain_v4/v4/lib",
+    "external/mongo_toolchain_v4/v4/lib64",
+]
+
+COMMON_BUILTIN_INCLUDE_DIRECTORIES = [
+    "/usr/include/openssl",
+    "/usr/include/{arch}-mongodb-linux",
+    "/usr/include",
+]
+
+COMMON_INCLUDE_DIRECTORIES = [
+    "external/mongo_toolchain_v4/stow/gcc-v4/include/c++/11.3.0",
+    "external/mongo_toolchain_v4/stow/gcc-v4/include/c++/11.3.0/{arch}-mongodb-linux",
+]
+
+COMMON_BINDIRS = [
+    "external/mongo_toolchain_v4/v4/bin",
+    "external/mongo_toolchain_v4/stow/gcc-v4/libexec/gcc/{arch}-mongodb-linux/11.3.0",
+    "external/mongo_toolchain_v4/stow/gcc-v4/lib/gcc/{arch}-mongodb-linux/11.3.0",
+    "external/mongo_toolchain_v4/stow/llvm-v4/bin",
+]
+
+GCC_INCLUDE_DIRS = [
+    "external/mongo_toolchain_v4/stow/gcc-v4/lib/gcc/{arch}-mongodb-linux/11.3.0/include",
+    "external/mongo_toolchain_v4/stow/gcc-v4/lib/gcc/{arch}-mongodb-linux/11.3.0/include-fixed",
+]
+
+CLANG_INCLUDE_DIRS = [
+    "external/mongo_toolchain_v4/stow/gcc-v4/include/c++/11.3.0/backward",
+    "external/mongo_toolchain_v4/stow/llvm-v4/lib/clang/12.0.1/include",
+]
\ No newline at end of file
diff --git a/bazel/toolchains/cc/mongo_linux/mongo_toolchain_version.bzl b/bazel/toolchains/cc/mongo_linux/mongo_toolchain_version.bzl
index a95664298fe..6609dd342c1 100644
--- a/bazel/toolchains/cc/mongo_linux/mongo_toolchain_version.bzl
+++ b/bazel/toolchains/cc/mongo_linux/mongo_toolchain_version.bzl
@@ -1,5 +1,5 @@
-load("//bazel/toolchains/cc/mongo_linux:mongo_toolchain_version_v5.bzl", "TOOLCHAIN_MAP_V5")
+load("//bazel/toolchains/cc/mongo_linux:mongo_toolchain_version_v4.bzl", "TOOLCHAIN_MAP_V4")
 
 TOOLCHAIN_MAP = {
-    "v5": TOOLCHAIN_MAP_V5,
+    "v4": TOOLCHAIN_MAP_V4,
 }
diff --git a/bazel/toolchains/cc/mongo_linux/mongo_toolchain_version_v4.bzl b/bazel/toolchains/cc/mongo_linux/mongo_toolchain_version_v4.bzl
new file mode 100644
index 00000000000..262ea10fb95
--- /dev/null
+++ b/bazel/toolchains/cc/mongo_linux/mongo_toolchain_version_v4.bzl
@@ -0,0 +1,115 @@
+# Use mongo/bazel/toolchains/mongo_toolchain_version_generator.py to generate this mapping for a given patch build.
+
+TOOLCHAIN_ID = "84bea98f485ef8c1af3f0612c56423fac4ea6256"
+TOOLCHAIN_MAP_V4 = {
+    "amazon_linux_2023_aarch64": {
+        "platform_name": "amazon2023-arm64",
+        "sha": "8da5d6be6439f3753da0ccb7cbbc312571947cae4c702cca4d64e58e3f659831",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-amazon2023-arm64-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "amazon_linux_2023_x86_64": {
+        "platform_name": "amazon2023",
+        "sha": "3513bf41e2b10b500ab346194dc5a803cc3cb4b66e7b74fdd8c6d76655e9116e",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-amazon2023-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "amazon_linux_2_aarch64": {
+        "platform_name": "amazon2-arm64",
+        "sha": "38bee5fa50ed1c8c527edfe511a652205b2ac5519a5ff30aaed67608749a8d31",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-amazon2-arm64-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "amazon_linux_2_x86_64": {
+        "platform_name": "amazon2",
+        "sha": "61edcd47f2f6c177801617f1ce1be168bb7b70003f42431da09c0aa4ea05b527",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-amazon2-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "debian10_x86_64": {
+        "platform_name": "debian10",
+        "sha": "901199126d6a10d77537fdcc19332588e679da589ab71a6500c71bcd2a48167d",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-debian10-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "debian12_x86_64": {
+        "platform_name": "debian12",
+        "sha": "40e94b657eddbe3c8cf89678e51c1a6f34d4a8cd73335962aa909ebed62270e3",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-debian12-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "rhel8_aarch64": {
+        "platform_name": "rhel82-arm64",
+        "sha": "1c5ffab1382ee37c6f3b64ee9b908b56f7fafb5ee413068a312cbae6c9ddbb47",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-rhel82-arm64-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "rhel8_ppc64le": {
+        "platform_name": "rhel81-ppc64le",
+        "sha": "baa9c82fe02c41a2fcb142848bd55f584bd85030217ec151b363a499a4b7565e",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-rhel81-ppc64le-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "rhel8_s390x": {
+        "platform_name": "rhel80-zseries",
+        "sha": "b45a31e88b8593290bfdaddfa6f81b518085096fa58f2dc6afab75ec4a23224f",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-rhel80-zseries-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "rhel8_x86_64": {
+        "platform_name": "rhel80",
+        "sha": "168aaf1a31eb9ebaad6b2e82f234c9ec1018fe87215ea832e2a973eb3dab9026",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-rhel80-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "rhel9_aarch64": {
+        "platform_name": "rhel90-arm64",
+        "sha": "85e61e65d3b8f4d469224ef6d8497a628173f76a8b6e2c534c81fb96a6728487",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-rhel90-arm64-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "rhel9_ppc64le": {
+        "platform_name": "rhel90-ppc64le",
+        "sha": "9fb1f50358d2910dd8cc675c107b206ac8f06d7d490fb75aa4a0ee06dc113534",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-rhel90-ppc64le-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "rhel9_s390x": {
+        "platform_name": "rhel90-zseries",
+        "sha": "9419449c6221a7d6ef084a6a0553e64d324b7adbddf6c8f664a40fb65cf6657e",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-rhel90-zseries-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "rhel9_x86_64": {
+        "platform_name": "rhel90",
+        "sha": "dbb449ece5981817e60f22fc72ac467991b17db6316bfeec655131d784181ef3",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-rhel90-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "suse15_x86_64": {
+        "platform_name": "suse15",
+        "sha": "9464edde3999d4456cbcea0085c52b027547c3d8b52c17a408fe191ae26dabd5",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-suse15-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "ubuntu18_x86_64": {
+        "platform_name": "ubuntu1804",
+        "sha": "7c8f9e3e889ec0d32149ce872bedfce4e0c08a71741513a92d65993a7ecf131a",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-ubuntu1804-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "ubuntu20_aarch64": {
+        "platform_name": "ubuntu2004-arm64",
+        "sha": "53caed99e13984bdcd9b666d41daece7e2c35d32dce1d994934ea5399cb6b6b0",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-ubuntu2004-arm64-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "ubuntu20_x86_64": {
+        "platform_name": "ubuntu2004",
+        "sha": "9ac5e210b779ace2ae4fa9b4e9911f3c5cb8f20be81e87fa1ae129fcd93af798",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-ubuntu2004-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "ubuntu22_aarch64": {
+        "platform_name": "ubuntu2204-arm64",
+        "sha": "eabe3ab69318800739d35de7bce772977ac447818d21043f77edea66a6b0730b",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-ubuntu2204-arm64-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "ubuntu22_x86_64": {
+        "platform_name": "ubuntu2204",
+        "sha": "70769f8a36b32955b4fb4af5307ac50d4dc9813228befe2aa194e604d9f7cfba",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-ubuntu2204-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "ubuntu24_aarch64": {
+        "platform_name": "ubuntu2404-arm64",
+        "sha": "ba617414633dce1ab5882337775ba0382a13ab9ea7138f82c7fb6519488822fc",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-ubuntu2404-arm64-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+    "ubuntu24_x86_64": {
+        "platform_name": "ubuntu2404",
+        "sha": "fdd1930124357b255e60dbace9fcbe667d1042d573d50727633d26ed26ad973c",
+        "url": "https://s3.amazonaws.com/boxes.10gen.com/build/toolchain/bazel_v4_toolchain-ubuntu2404-84bea98f485ef8c1af3f0612c56423fac4ea6256.tar.gz",
+    },
+}
diff --git a/bazel/toolchains/cc/mongo_toolchain_version.bzl b/bazel/toolchains/cc/mongo_toolchain_version.bzl
index 85362f52a30..548d92495ec 100644
--- a/bazel/toolchains/cc/mongo_toolchain_version.bzl
+++ b/bazel/toolchains/cc/mongo_toolchain_version.bzl
@@ -1,5 +1,5 @@
-load("//bazel/toolchains/cc:mongo_toolchain_version_v5.bzl", "TOOLCHAIN_MAP_V5")
+load("//bazel/toolchains/cc:mongo_toolchain_version_v4.bzl", "TOOLCHAIN_MAP_V4")
 
 TOOLCHAIN_MAP = {
-    "v5": TOOLCHAIN_MAP_V5,
+    "v4": TOOLCHAIN_MAP_V4,
 }
diff --git a/src/third_party/unwind/BUILD.bazel b/src/third_party/unwind/BUILD.bazel
index 57a1c15a2ea..e29d6576439 100644
--- a/src/third_party/unwind/BUILD.bazel
+++ b/src/third_party/unwind/BUILD.bazel
@@ -173,7 +173,6 @@ mongo_cc_library(
     }) + select({
         "@//bazel/config:compiler_type_clang": [
             "-Wno-header-guard",
-            "-Wno-single-bit-bitfield-constant-conversion",
         ],
         "@//bazel/config:compiler_type_gcc": ["-Wno-uninitialized"],
         "//conditions:default": [],
