diff --git a/buildscripts/unittest_grouper.py b/buildscripts/unittest_grouper.py
index f1db2064f61..a18f20cf3bf 100644
--- a/buildscripts/unittest_grouper.py
+++ b/buildscripts/unittest_grouper.py
@@ -12,11 +12,10 @@ from typing import Dict, List
 
 from retry import retry
 
-from buildscripts.simple_report import make_report, put_report, try_combine_reports
-
 sys.path.append(".")
 
 from buildscripts.install_bazel import install_bazel
+from buildscripts.simple_report import make_report, put_report, try_combine_reports
 
 RELEASE_URL = "https://github.com/bazelbuild/buildtools/releases/download/v7.3.1/"
 
