diff --git a/bazel/mongo_src_rules.bzl b/bazel/mongo_src_rules.bzl
index 81f281e4855..64a17e5aeda 100644
--- a/bazel/mongo_src_rules.bzl
+++ b/bazel/mongo_src_rules.bzl
@@ -486,7 +486,10 @@ def mongo_cc_library(
             "//conditions:default": ["-Wl,-z,defs"],
         })
     else:
-        undefined_ref_flag = []
+        undefined_ref_flag = select({
+            "@platforms//os:macos": ["-Wl,-undefined,dynamic_lookup"],
+            "//conditions:default": [],
+        })
 
     create_header_dep(
         name = name + HEADER_DEP_SUFFIX,
