diff --git a/BUILD.bazel b/BUILD.bazel
index 178508a3abe..5e3a07fc867 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -266,7 +266,7 @@ mongo_install(
         "//src/mongo/db:mongotrafficreader",
         "//src/mongo/db/query/search/mongotmock",
         "//src/mongo/db/query/stage_builder/sbe/abt:optimizer_gdb_test_program",
-        "//src/mongo/db/storage/key_string:ksdecode",
+        "//src/mongo/db/storage:ksdecode",
         "//src/mongo/s:mongos",
         "//src/mongo/shell:mongo",
         "//src/mongo/tools/mongobridge_tool:mongobridge",
