diff --git a/BUILD.bazel b/BUILD.bazel
index deec32d6ead..a04b4457859 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -282,7 +282,7 @@ mongo_install(
     pretty_printer_tests = {
         "//src/mongo/util:pretty_printer_test.py": "//src/mongo/util:pretty_printer_test_program",
         "//src/mongo/db/concurrency:lock_gdb_test.py": "//src/mongo/db:mongod",
-        "//src/mongo/db/query/stage_builder/sbe/abt:optimizer_gdb_test.py": "//src/mongo/db/query/stage_builder/sbe/abt:optimizer_gdb_test_program",
+        "//src/mongo/db/query/optimizer:optimizer_gdb_test.py": "//src/mongo/db/query/optimizer:optimizer_gdb_test_program",
     },
     deps = [
         "//src/mongo/db/modules/enterprise:dist-test",
