diff --git a/MODULE.bazel b/MODULE.bazel
index 48a65c7957a..7f93d3e102d 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -28,16 +28,3 @@ use_repo(cc_configure, "local_config_cc")
 bazel_dep(name = "rules_cc", version = "0.1.1")
 bazel_dep(name = "bazel_skylib", version = "1.7.1")
 bazel_dep(name = "platforms", version = "0.0.11")
-
-bazel_dep(
-    name = "google_benchmark",
-    version = "1.9.2",
-    dev_dependency = True,
-)
-
-# Note: Googletest is NOT a dev_dependency. Some Abseil test utilities
-# intended to be used by Abseil users depend on GoogleTest.
-bazel_dep(
-    name = "googletest",
-    version = "1.17.0",
-)
