diff --git a/absl/debugging/symbolize_win32.inc b/absl/debugging/symbolize_win32.inc
index b3c7b0be..800993a3 100644
--- a/absl/debugging/symbolize_win32.inc
+++ b/absl/debugging/symbolize_win32.inc
@@ -42,7 +42,11 @@ void InitializeSymbolizer(const char*) {
   if (process != nullptr) {
     return;
   }
-  process = GetCurrentProcess();
+  HANDLE currentProcess = GetCurrentProcess();
+  if(!DuplicateHandle(currentProcess, currentProcess, currentProcess, &process, 0, FALSE, DUPLICATE_SAME_ACCESS)) {
+    const unsigned long long error{GetLastError()};
+    ABSL_RAW_LOG(FATAL, "SymInitialize() failed: %llu", error);
+  }
 
   // Symbols are not loaded until a reference is made requiring the
   // symbols be loaded. This is the fastest, most efficient way to use
