From 1596d50ad9770171539f1e07643bf1fd401a8872 Mon Sep 17 00:00:00 2001
From: Billy Donahue <billy.donahue@mongodb.com>
Date: Wed, 19 Dec 2018 11:16:51 -0500
Subject: [PATCH] ABSL_FORCE_ALIGNED_ACCESS

---
 absl/base/internal/unaligned_access.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/absl/base/internal/unaligned_access.h b/absl/base/internal/unaligned_access.h
index f9df3b7..d2aa885 100644
--- a/absl/base/internal/unaligned_access.h
+++ b/absl/base/internal/unaligned_access.h
@@ -108,7 +108,7 @@ inline void UnalignedStore64(void *p, uint64_t v) {
 #define ABSL_INTERNAL_UNALIGNED_STORE64(_p, _val) \
   (absl::base_internal::UnalignedStore64(_p, _val))
 
-#elif defined(UNDEFINED_BEHAVIOR_SANITIZER)
+#elif defined(UNDEFINED_BEHAVIOR_SANITIZER) || defined(ABSL_FORCE_ALIGNED_ACCESS)
 
 namespace absl {
 namespace base_internal {
-- 
2.17.1

