From e88fe4c225e9efe5c50d49e2a697edeb20d9ed8c Mon Sep 17 00:00:00 2001
From: Henrik Edin <henrik.edin@mongodb.com>
Date: Thu, 8 Nov 2018 18:32:09 -0500
Subject: [PATCH 2/2] Use _umul128 on Windows to improve performance of Mix()
 on MSVC.

---
 absl/hash/internal/hash.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/absl/hash/internal/hash.h b/absl/hash/internal/hash.h
index b66cc1b..7106077 100644
--- a/absl/hash/internal/hash.h
+++ b/absl/hash/internal/hash.h
@@ -773,6 +773,7 @@ class CityHashState : public HashStateBase<CityHashState> {
   }
 
   ABSL_ATTRIBUTE_ALWAYS_INLINE static uint64_t Mix(uint64_t state, uint64_t v) {
+#if !defined(_MSC_VER) || !defined(_WIN64)
     using MultType =
         absl::conditional_t<sizeof(size_t) == 4, uint64_t, uint128>;
     // We do the addition in 64-bit space to make sure the 128-bit
@@ -782,6 +783,11 @@ class CityHashState : public HashStateBase<CityHashState> {
     MultType m = state + v;
     m *= kMul;
     return static_cast<uint64_t>(m ^ (m >> (sizeof(m) * 8 / 2)));
+#else
+    uint64_t high;
+    uint64_t low = _umul128(state + v, kMul, &high);
+    return low ^ high;
+#endif
   }
 
   // Seed()
-- 
2.10.1.windows.1

