diff --git a/src/mongo/util/jemalloc_server_status_section.cpp b/src/mongo/util/jemalloc_server_status_section.cpp
new file mode 100644
index 00000000000..27384dc6001
--- /dev/null
+++ b/src/mongo/util/jemalloc_server_status_section.cpp
@@ -0,0 +1,98 @@
+/**
+ *    Copyright (C) 2023-present MongoDB, Inc.
+ *
+ *    This program is free software: you can redistribute it and/or modify
+ *    it under the terms of the Server Side Public License, version 1,
+ *    as published by MongoDB, Inc.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    Server Side Public License for more details.
+ *
+ *    You should have received a copy of the Server Side Public License
+ *    along with this program. If not, see
+ *    <http://www.mongodb.com/licensing/server-side-public-license>.
+ *
+ *    As a special exception, the copyright holders give permission to link the
+ *    code of portions of this program with the OpenSSL library under certain
+ *    conditions as described in each individual source file and distribute
+ *    linked combinations including the program with the OpenSSL library. You
+ *    must comply with the Server Side Public License in all respects for
+ *    all of the code used other than as permitted herein. If you modify file(s)
+ *    with this exception, you may extend this exception to your version of the
+ *    file(s), but you are not obligated to do so. If you do not wish to do so,
+ *    delete this exception statement from your version. If you delete this
+ *    exception statement from all source files in the program, then also delete
+ *    it in the license file.
+ */
+
+#define MONGO_LOGV2_DEFAULT_COMPONENT ::mongo::logv2::LogComponent::kDefault
+
+#include "mongo/platform/basic.h"
+
+#include "jemalloc/jemalloc.h"
+
+#include "mongo/base/init.h"
+#include "mongo/db/commands/server_status.h"
+#include "mongo/logv2/log.h"
+
+namespace mongo {
+
+namespace {
+
+class JEMallocServerStatusSection : public ServerStatusSection {
+public:
+    JEMallocServerStatusSection() : ServerStatusSection("jemalloc") {}
+    virtual bool includeByDefault() const {
+        return true;
+    }
+
+    virtual BSONObj generateSection(OperationContext* txn, const BSONElement& configElement) const {
+        long long verbosity = 1;
+        if (configElement) {
+            // Relies on the fact that safeNumberLong turns non-numbers into 0.
+            long long configValue = configElement.safeNumberLong();
+            if (configValue) {
+                verbosity = configValue;
+            }
+        }
+
+        BSONObjBuilder builder;
+
+        // For a list of properties see the jemalloc man page
+        {
+            BSONObjBuilder sub(builder.subobjStart("generic"));
+            appendNumericPropertyIfAvailable(sub, "current_allocated_bytes", "stats.allocated");
+            /* TODO: I think this is right? */
+            appendNumericPropertyIfAvailable(sub, "heap_size", "stats.active");
+        }
+        {
+            BSONObjBuilder sub(builder.subobjStart("jemalloc"));
+
+            appendNumericPropertyIfAvailable(sub, "allocated_chunk_highwater", "stats.chunks.high");
+            appendNumericPropertyIfAvailable(sub, "allocated_huge", "stats.huge.allocated");
+            appendNumericPropertyIfAvailable(sub, "allocated_huge_highwater", "stats.huge.nmalloc");
+
+            /* TODO: Should switch this to malloc_stats_print
+                char buffer[4096];
+                MallocExtension::instance()->GetStats(buffer, sizeof buffer);
+                builder.append("formattedString", buffer);
+            */
+        }
+
+        return builder.obj();
+    }
+
+private:
+    static void appendNumericPropertyIfAvailable(BSONObjBuilder& builder,
+                                                 StringData bsonName,
+                                                 const char* property) {
+        size_t sz, value;
+        if (mallctl(property, &value, &sz, NULL, 0) == 0)
+            builder.appendNumber(bsonName, static_cast<long long>(value));
+    }
+
+} jemallocServerStatusSection;
+}  // namespace
+}  // namespace mongo
diff --git a/src/mongo/util/jemalloc_set_parameter.cpp b/src/mongo/util/jemalloc_set_parameter.cpp
new file mode 100644
index 00000000000..0cc03df9ef6
--- /dev/null
+++ b/src/mongo/util/jemalloc_set_parameter.cpp
@@ -0,0 +1,139 @@
+/**
+ *    Copyright (C) 2023-present MongoDB, Inc.
+ *
+ *    This program is free software: you can redistribute it and/or modify
+ *    it under the terms of the Server Side Public License, version 1,
+ *    as published by MongoDB, Inc.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    Server Side Public License for more details.
+ *
+ *    You should have received a copy of the Server Side Public License
+ *    along with this program. If not, see
+ *    <http://www.mongodb.com/licensing/server-side-public-license>.
+ *
+ *    As a special exception, the copyright holders give permission to link the
+ *    code of portions of this program with the OpenSSL library under certain
+ *    conditions as described in each individual source file and distribute
+ *    linked combinations including the program with the OpenSSL library. You
+ *    must comply with the Server Side Public License in all respects for
+ *    all of the code used other than as permitted herein. If you modify file(s)
+ *    with this exception, you may extend this exception to your version of the
+ *    file(s), but you are not obligated to do so. If you do not wish to do so,
+ *    delete this exception statement from your version. If you delete this
+ *    exception statement from all source files in the program, then also delete
+ *    it in the license file.
+ */
+
+#include "mongo/platform/basic.h"
+
+#include "jemalloc/jemalloc.h"
+#include <algorithm>
+
+#include "mongo/base/init.h"
+#include "mongo/base/parse_number.h"
+#include "mongo/base/status.h"
+#include "mongo/db/jsobj.h"
+#include "mongo/idl/server_parameter.h"
+#include "mongo/platform/atomic_proxy.h"
+#include "mongo/platform/atomic_word.h"
+#include "mongo/util/processinfo.h"
+
+namespace mongo {
+namespace {
+
+class JemallocNumericPropertyServerParameter : public ServerParameter {
+    JemallocNumericPropertyServerParameter(const JemallocNumericPropertyServerParameter&) = delete;
+    JemallocNumericPropertyServerParameter& operator=(
+        const JemallocNumericPropertyServerParameter&) = delete;
+
+public:
+    explicit JemallocNumericPropertyServerParameter(const std::string& serverParameterName,
+                                                    const std::string& jemallocPropertyName);
+
+    virtual void append(OperationContext* txn, BSONObjBuilder& b, const std::string& name);
+    virtual Status set(const BSONElement& newValueElement);
+    virtual Status setFromString(const std::string& str);
+
+private:
+    const std::string _jemallocPropertyName;
+};
+
+JemallocNumericPropertyServerParameter::JemallocNumericPropertyServerParameter(
+    const std::string& serverParameterName, const std::string& jemallocPropertyName)
+    : ServerParameter(serverParameterName, ServerParameterType::kStartupAndRuntime),
+      _jemallocPropertyName(jemallocPropertyName) {}
+
+void JemallocNumericPropertyServerParameter::append(OperationContext* txn,
+                                                    BSONObjBuilder& b,
+                                                    const std::string& name) {
+    size_t sz, value;
+    if (mallctl(_jemallocPropertyName.c_str(), &value, &sz, NULL, 0)) {
+        b.appendNumber(name, static_cast<long long>(value));
+    }
+}
+
+Status JemallocNumericPropertyServerParameter::set(const BSONElement& newValueElement) {
+    if (!newValueElement.isNumber()) {
+        return Status(ErrorCodes::TypeMismatch,
+                      str::stream()
+                          << "Expected server parameter " << newValueElement.fieldName()
+                          << " to have numeric type, but found " << newValueElement.toString(false)
+                          << " of type " << typeName(newValueElement.type()));
+    }
+    long long valueAsLongLong = newValueElement.safeNumberLong();
+    if (valueAsLongLong < 0 ||
+        static_cast<unsigned long long>(valueAsLongLong) > std::numeric_limits<size_t>::max()) {
+        return Status(ErrorCodes::BadValue,
+                      str::stream()
+                          << "Value " << newValueElement.toString(false) << " is out of range for "
+                          << newValueElement.fieldName() << "; expected a value between 0 and "
+                          << std::min<unsigned long long>(std::numeric_limits<size_t>::max(),
+                                                          std::numeric_limits<long long>::max()));
+    }
+    size_t valueAsSizet = static_cast<size_t>(valueAsLongLong);
+    if (!mallctl(_jemallocPropertyName.c_str(), NULL, 0, &valueAsSizet, sizeof(size_t))) {
+        return Status(ErrorCodes::InternalError,
+                      str::stream()
+                          << "Failed to set internal jemalloc property " << _jemallocPropertyName);
+    }
+    return Status::OK();
+}
+
+Status JemallocNumericPropertyServerParameter::setFromString(const std::string& str) {
+    long long valueAsLongLong;
+    Status status = NumberParser{}(str, &valueAsLongLong);
+    if (!status.isOK()) {
+        return status;
+    }
+    BSONObjBuilder builder;
+    builder.append(name(), valueAsLongLong);
+    return set(builder.done().firstElement());
+}
+
+// JEMalloc uses a certain number of arenas, shared amongst all threads. The default value is
+// eight times the number of CPUs, which can contribute to fragmentation.
+JemallocNumericPropertyServerParameter jemallocMaxArenas("jemallocMaxArenas", "opt.narenas");
+
+JemallocNumericPropertyServerParameter jemallocRedzone("jemallocRedzone", "opt.redzone");
+
+MONGO_SERVER_PARAMETER_REGISTER(JemallocServerParameters)(InitializerContext*) {
+    registerServerParameter(&jemallocMaxArenas);
+    registerServerParameter(&jemallocRedzone);
+}
+
+MONGO_INITIALIZER_GENERAL(JemallocConfigurationDefaults, (), ("BeginStartupOptionHandling"))
+(InitializerContext*) {
+    // Don't set our custom configuration options if the user has configured their own
+    if (getenv("MALLOC_CONF")) {
+        return;
+    }
+
+    uassertStatusOK(jemallocRedzone.setFromString("0"));
+    uassertStatusOK(jemallocMaxArenas.setFromString("8"));
+}
+
+}  // namespace
+}  // namespace mongo
