diff --git a/src/mongo/db/storage/wiredtiger/wiredtiger_session.h b/src/mongo/db/storage/wiredtiger/wiredtiger_session.h
index ea71e2c430a..9ea27bcc165 100644
--- a/src/mongo/db/storage/wiredtiger/wiredtiger_session.h
+++ b/src/mongo/db/storage/wiredtiger/wiredtiger_session.h
@@ -110,7 +110,6 @@ public:
     WRAPPED_WT_SESSION_METHOD(create)
     WRAPPED_WT_SESSION_METHOD(drop)
     WRAPPED_WT_SESSION_METHOD(get_last_error)
-    WRAPPED_WT_SESSION_METHOD(get_rollback_reason)
     WRAPPED_WT_SESSION_METHOD(log_flush)
     WRAPPED_WT_SESSION_METHOD(open_cursor)
     WRAPPED_WT_SESSION_METHOD(prepare_transaction)
