diff --git a/src/third_party/mozjs-38/extract/js/src/prmjtime.cpp b/src/third_party/mozjs-38/extract/js/src/prmjtime.cpp
index b5a454a..2372791 100644
--- a/src/third_party/mozjs-38/extract/js/src/prmjtime.cpp
+++ b/src/third_party/mozjs-38/extract/js/src/prmjtime.cpp
@@ -133,7 +133,7 @@ PRMJ_NowInit()
     InitializeCriticalSectionAndSpinCount(&calibration.data_lock, DataLockSpinCount);
 
     // Windows 8 has a new API function we can use.
-    if (HMODULE h = GetModuleHandle(L"kernel32.dll")) {
+    if (HMODULE h = GetModuleHandle("kernel32.dll")) {
         pGetSystemTimePreciseAsFileTime =
             (void (WINAPI*)(LPFILETIME))GetProcAddress(h, "GetSystemTimePreciseAsFileTime");
     }
diff --git a/src/third_party/mozjs-38/extract/mfbt/lz4.c b/src/third_party/mozjs-38/extract/mfbt/lz4.c
index e05665f..042a220 100644
--- a/src/third_party/mozjs-38/extract/mfbt/lz4.c
+++ b/src/third_party/mozjs-38/extract/mfbt/lz4.c
@@ -877,7 +877,7 @@ FORCE_INLINE int LZ4_decompress_generic(
                  )
 {
     /* Local Variables */
-    const BYTE* ip = (const BYTE*) source;
+    const BYTE* restrict ip = (const BYTE*) source;
     const BYTE* ref;
     const BYTE* const iend = ip + inputSize;
 
diff --git a/src/third_party/mozjs-38/include/mozilla/Types.h b/src/third_party/mozjs-38/include/mozilla/Types.h
index 2697ec1..e7e18ab 100644
--- a/src/third_party/mozjs-38/include/mozilla/Types.h
+++ b/src/third_party/mozjs-38/include/mozilla/Types.h
@@ -37,10 +37,7 @@
  * These macros are designed for use by library interfaces -- not for normal
  * methods or data used cross-file.
  */
-// MongoDB Modification - SERVER-20311
-// Disable all visibility hints since we are statically linking on Windows.
-// This prevents mongod.exe and other binaries from exporting these functions.
-#if 0 //defined(WIN32)
+#if defined(WIN32)
 #  define MOZ_EXPORT   __declspec(dllexport)
 #else /* Unix */
 #  ifdef HAVE_VISIBILITY_ATTRIBUTE
@@ -52,6 +49,7 @@
 #  endif
 #endif
 
+
 /*
  * Whereas implementers use MOZ_EXPORT to declare and define library symbols,
  * users use MOZ_IMPORT_API and MOZ_IMPORT_DATA to access them.  Most often the
