diff --git a/src/third_party/opentelemetry-cpp/api/include/opentelemetry/baggage/baggage.h b/src/third_party/opentelemetry-cpp/api/include/opentelemetry/baggage/baggage.h
index 6c799cda278..d6ee7435a8e 100644
--- a/src/third_party/opentelemetry-cpp/api/include/opentelemetry/baggage/baggage.h
+++ b/src/third_party/opentelemetry-cpp/api/include/opentelemetry/baggage/baggage.h
@@ -36,8 +36,9 @@ public:
 
   OPENTELEMETRY_API_SINGLETON static nostd::shared_ptr<Baggage> GetDefault()
   {
-    static nostd::shared_ptr<Baggage> baggage{new Baggage()};
-    return baggage;
+    // intentionally leak
+    static auto baggage = new nostd::shared_ptr<Baggage>{new Baggage()};
+    return *baggage;
   }
 
   /* Get value for key in the baggage
diff --git a/src/third_party/opentelemetry-cpp/api/include/opentelemetry/baggage/baggage_context.h b/src/third_party/opentelemetry-cpp/api/include/opentelemetry/baggage/baggage_context.h
index a0016353ba6..2545105eff0 100644
--- a/src/third_party/opentelemetry-cpp/api/include/opentelemetry/baggage/baggage_context.h
+++ b/src/third_party/opentelemetry-cpp/api/include/opentelemetry/baggage/baggage_context.h
@@ -22,8 +22,9 @@ inline nostd::shared_ptr<Baggage> GetBaggage(const context::Context &context) no
   {
     return nostd::get<nostd::shared_ptr<Baggage>>(context_value);
   }
-  static nostd::shared_ptr<Baggage> empty_baggage{new Baggage()};
-  return empty_baggage;
+  // intentionally leak
+  static auto empty_baggage = new nostd::shared_ptr<Baggage>{new Baggage()};
+  return *empty_baggage;
 }
 
 inline context::Context SetBaggage(context::Context &context,
diff --git a/src/third_party/opentelemetry-cpp/api/include/opentelemetry/context/propagation/global_propagator.h b/src/third_party/opentelemetry-cpp/api/include/opentelemetry/context/propagation/global_propagator.h
index 85293202f69..f4e543bd578 100644
--- a/src/third_party/opentelemetry-cpp/api/include/opentelemetry/context/propagation/global_propagator.h
+++ b/src/third_party/opentelemetry-cpp/api/include/opentelemetry/context/propagation/global_propagator.h
@@ -41,14 +41,16 @@ public:
 private:
   OPENTELEMETRY_API_SINGLETON static nostd::shared_ptr<TextMapPropagator> &GetPropagator() noexcept
   {
-    static nostd::shared_ptr<TextMapPropagator> propagator(new NoOpPropagator());
-    return propagator;
+    // intentionally leak
+    static auto propagator = new nostd::shared_ptr<TextMapPropagator>(new NoOpPropagator());
+    return *propagator;
   }
 
   OPENTELEMETRY_API_SINGLETON static common::SpinLockMutex &GetLock() noexcept
   {
-    static common::SpinLockMutex lock;
-    return lock;
+    // intentionally leak
+    static auto lock = new common::SpinLockMutex();
+    return *lock;
   }
 };
 
diff --git a/src/third_party/opentelemetry-cpp/api/include/opentelemetry/context/runtime_context.h b/src/third_party/opentelemetry-cpp/api/include/opentelemetry/context/runtime_context.h
index 0ef170571b5..d0605009d7b 100644
--- a/src/third_party/opentelemetry-cpp/api/include/opentelemetry/context/runtime_context.h
+++ b/src/third_party/opentelemetry-cpp/api/include/opentelemetry/context/runtime_context.h
@@ -178,8 +178,9 @@ private:
 
   OPENTELEMETRY_API_SINGLETON static nostd::shared_ptr<RuntimeContextStorage> &GetStorage() noexcept
   {
-    static nostd::shared_ptr<RuntimeContextStorage> context(GetDefaultStorage());
-    return context;
+    // intentionally leak
+    static auto context = new nostd::shared_ptr<RuntimeContextStorage>(GetDefaultStorage());
+    return *context;
   }
 };
 
diff --git a/src/third_party/opentelemetry-cpp/api/include/opentelemetry/metrics/provider.h b/src/third_party/opentelemetry-cpp/api/include/opentelemetry/metrics/provider.h
index 20a606a772e..3b49eb1d394 100644
--- a/src/third_party/opentelemetry-cpp/api/include/opentelemetry/metrics/provider.h
+++ b/src/third_party/opentelemetry-cpp/api/include/opentelemetry/metrics/provider.h
@@ -45,14 +45,16 @@ public:
 private:
   OPENTELEMETRY_API_SINGLETON static nostd::shared_ptr<MeterProvider> &GetProvider() noexcept
   {
-    static nostd::shared_ptr<MeterProvider> provider(new NoopMeterProvider);
-    return provider;
+    // intentionally leak
+    static auto provider = new nostd::shared_ptr<MeterProvider>(new NoopMeterProvider);
+    return *provider;
   }
 
   OPENTELEMETRY_API_SINGLETON static common::SpinLockMutex &GetLock() noexcept
   {
-    static common::SpinLockMutex lock;
-    return lock;
+    // intentionally leak
+    static auto lock = new common::SpinLockMutex();
+    return *lock;
   }
 };
 
diff --git a/src/third_party/opentelemetry-cpp/api/include/opentelemetry/trace/noop.h b/src/third_party/opentelemetry-cpp/api/include/opentelemetry/trace/noop.h
index 14e7c9c33d5..d6eb2d3c6a5 100644
--- a/src/third_party/opentelemetry-cpp/api/include/opentelemetry/trace/noop.h
+++ b/src/third_party/opentelemetry-cpp/api/include/opentelemetry/trace/noop.h
@@ -110,9 +110,11 @@ public:
   {
     // Don't allocate a no-op span for every StartSpan call, but use a static
     // singleton for this case.
-    static nostd::shared_ptr<trace::Span> noop_span(new trace::NoopSpan{this->shared_from_this()});
+    // intentionally leak
+    static auto noop_span =
+        new nostd::shared_ptr<trace::Span>(new trace::NoopSpan{this->shared_from_this()});
 
-    return noop_span;
+    return *noop_span;
   }
 
 #if OPENTELEMETRY_ABI_VERSION_NO == 1
diff --git a/src/third_party/opentelemetry-cpp/api/include/opentelemetry/trace/provider.h b/src/third_party/opentelemetry-cpp/api/include/opentelemetry/trace/provider.h
index 7e22dc69f54..a0e7ce52dc3 100644
--- a/src/third_party/opentelemetry-cpp/api/include/opentelemetry/trace/provider.h
+++ b/src/third_party/opentelemetry-cpp/api/include/opentelemetry/trace/provider.h
@@ -45,14 +45,16 @@ public:
 private:
   OPENTELEMETRY_API_SINGLETON static nostd::shared_ptr<TracerProvider> &GetProvider() noexcept
   {
-    static nostd::shared_ptr<TracerProvider> provider(new NoopTracerProvider);
-    return provider;
+    // intentionally leak
+    static auto provider = new nostd::shared_ptr<TracerProvider>(new NoopTracerProvider);
+    return *provider;
   }
 
   OPENTELEMETRY_API_SINGLETON static common::SpinLockMutex &GetLock() noexcept
   {
-    static common::SpinLockMutex lock;
-    return lock;
+    // intentionally leak
+    static auto lock = new common::SpinLockMutex();
+    return *lock;
   }
 };
 
diff --git a/src/third_party/opentelemetry-cpp/api/include/opentelemetry/trace/trace_state.h b/src/third_party/opentelemetry-cpp/api/include/opentelemetry/trace/trace_state.h
index df7a0e0fe9c..41e558952bf 100644
--- a/src/third_party/opentelemetry-cpp/api/include/opentelemetry/trace/trace_state.h
+++ b/src/third_party/opentelemetry-cpp/api/include/opentelemetry/trace/trace_state.h
@@ -42,8 +42,9 @@ public:
 
   OPENTELEMETRY_API_SINGLETON static nostd::shared_ptr<TraceState> GetDefault()
   {
-    static nostd::shared_ptr<TraceState> ts{new TraceState()};
-    return ts;
+    // intentionally leak
+    static auto ts = new nostd::shared_ptr<TraceState>{new TraceState()};
+    return *ts;
   }
 
   /**
