diff -ru mongo-c-driver-1.9.2/CMakeLists.txt mongo-c-driver/CMakeLists.txt
Once CDRIVER-2631 and CDRIVER-2630 are released, we can remove this patch
--- mongo-c-driver-1.9.2/CMakeLists.txt	2018-01-08 09:15:49.000000000 -0500
+++ mongo-c-driver/CMakeLists.txt	2018-05-01 10:52:32.346989684 -0400
@@ -199,9 +199,12 @@
 endif ()
 
 include(CheckLibraryExists)
-CHECK_LIBRARY_EXISTS(ssl ASN1_STRING_get0_data "openssl/asn1.h" MONGOC_HAVE_ASN1_STRING_GET0_DATA)
-if (NOT MONGOC_HAVE_ASN1_STRING_GET0_DATA)
-   set (MONGOC_HAVE_ASN1_STRING_GET0_DATA 0)
+set (MONGOC_HAVE_ASN1_STRING_GET0_DATA 0)
+if (NOT ENABLE_SSL STREQUAL OFF)
+    CHECK_LIBRARY_EXISTS(ssl ASN1_STRING_get0_data "openssl/asn1.h" MONGOC_HAVE_ASN1_STRING_GET0_DATA)
+    if (NOT MONGOC_HAVE_ASN1_STRING_GET0_DATA)
+       set (MONGOC_HAVE_ASN1_STRING_GET0_DATA 0)
+    endif ()
 endif ()
 
 if (NOT (ENABLE_SASL STREQUAL CYRUS
@@ -398,7 +401,7 @@
 include_directories("${PROJECT_BINARY_DIR}/src/mongoc")
 include_directories("${SOURCE_DIR}/src/mongoc")
 
-if (UNIX AND NOT APPLE)
+if (UNIX AND NOT APPLE AND NOT ANDROID)
    set(SHM_LIB -lrt)
 endif()
 
