--- mongo-c-driver-1.9.1/src/tools/mongoc-stat.c	2018-01-04 19:41:10.000000000 -0500
+++ mongo-c-driver/src/tools/mongoc-stat.c-new	2018-01-30 14:04:07.300797964 -0500
@@ -94,7 +94,11 @@
    snprintf (name, sizeof name, "/mongoc-%u", pid);
    name[sizeof name - 1] = '\0';
 
+#if defined(BSON_OS_UNIX) && defined(MONGOC_ENABLE_SHM_COUNTERS)
    if (-1 == (fd = shm_open (name, O_RDONLY, 0))) {
+#else
+   if (true) {
+#endif
       perror ("Failed to load shared memory segment");
       return NULL;
    }
